MicroChess/src/Renderer/sprite2d.cpp

```diff
@@ -326,27 +325,42 @@ namespace ChronoShift
     void RenderTextEntities()
     {
         FunctionQueue text_render_queue;
+        constexpr float FONT_SCALE_FACTOR = 3.0f;
 
         for (auto& txtentity : FlexECS::Scene::GetActiveScene()->CachedQuery<IsActive, ZIndex, Transform, Shader, Text>())
         {
             if (!txtentity.GetComponent<IsActive>()->is_active) continue;
 
-            Matrix4x4 transform = txtentity.GetComponent<Transform>()->transform;
+            Matrix4x4& transform = txtentity.GetComponent<Transform>()->transform;
             auto shader = FlexECS::Scene::GetActiveScene()->Internal_StringStorage_Get(txtentity.GetComponent<Shader>()->shader);
-            auto font = FlexECS::Scene::GetActiveScene()->Internal_StringStorage_Get(txtentity.GetComponent<Text>()->fonttype);
-
+            auto* textComponent = txtentity.GetComponent<Text>();
+            auto font = FlexECS::Scene::GetActiveScene()->Internal_StringStorage_Get(textComponent->fonttype);
+            
             Renderer2DText sample;
-            sample.m_words = FlexECS::Scene::GetActiveScene()->Internal_StringStorage_Get(txtentity.GetComponent<Text>()->text);
+            sample.m_words = FlexECS::Scene::GetActiveScene()->Internal_StringStorage_Get(textComponent->text);
             sample.m_shader = shader;
             sample.m_fonttype = font;
             sample.m_transform = transform;
+            sample.m_window_size = Vector2(static_cast<float>(FlexEngine::Application::GetCurrentWindow()->GetWidth()), static_cast<float>(FlexEngine::Application::GetCurrentWindow()->GetHeight()));
```

Comment:
Not important right now, but can subscribe to the resize window callback for GLFW